test/gpu/octree/test_radius_search.cpp

```diff
@@ -103,24 +103,34 @@ TEST(PCL_OctreeGPU, batchRadiusSearch)
         host_search1[i].reserve(max_answers);
         host_search2[i].reserve(max_answers);
     }    
-    
+
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances1;
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances2;
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances3;
+
     //search GPU shared
-    octree_device.radiusSearch(queries_device, data.shared_radius, max_answers, result_device1);
+    octree_device.radiusSearch(queries_device, data.shared_radius, max_answers, result_device1, result_sqr_distances1);
 
     //search GPU individual
-    octree_device.radiusSearch(queries_device,    radiuses_device, max_answers, result_device2);
+    octree_device.radiusSearch(queries_device,    radiuses_device, max_answers, result_device2, result_sqr_distances2);
 
     //search GPU shared with indices
     pcl::gpu::Octree::Indices indices;
     indices.upload(data.indices);
-    octree_device.radiusSearch(queries_device, indices, data.shared_radius, max_answers, result_device3);
+    octree_device.radiusSearch(queries_device, indices, data.shared_radius, max_answers, result_device3, result_sqr_distances3);
 
     //search CPU
     octree_device.internalDownload();
+
+    std::vector< std::vector<float> > host_sqr_distances1(data.tests_num);
+    std::vector< std::vector<float> > host_sqr_distances2(data.tests_num);
+
     for(std::size_t i = 0; i < data.tests_num; ++i)
     {
-        octree_device.radiusSearchHost(data.queries[i], data.shared_radius, host_search1[i], max_answers);
-        octree_device.radiusSearchHost(data.queries[i], data.radiuses[i],   host_search2[i], max_answers);
+        host_sqr_distances1[i].reserve(max_answers);
+        host_sqr_distances2[i].reserve(max_answers);
```

Comment:
I'm reserving memory for the out variable here, which goes against @SergioRAgostinho 's comments on #4322, I followed this approach because the existing API requires the user to allocate memory for the other out varaible (for result indices) anyway. Should I change this so that the allocation happens inside the method?

Reply:
Yes please.